test/integration/parachain/staging/sequential/loans.test.ts

```diff
@@ -354,8 +353,45 @@ describe("Loans", () => {
     });
 
     describe("getAccruedRewardsOfAccount", () => {
-        it("should return correct amount of reward", () => {
-            // TODO
+        before(async function () {
+            const addRewardExtrinsic = sudoInterBtcAPI.api.tx.loans.addReward("100000000000000");
+            const updateRewardSpeedExtrinsic_1 = sudoInterBtcAPI.api.tx.loans.updateMarketRewardSpeed(
+                underlyingCurrencyId,
+                "1000000000000",
+                "0"
+            );
+            const updateRewardSpeedExtrinsic_2 = sudoInterBtcAPI.api.tx.loans.updateMarketRewardSpeed(
+                underlyingCurrencyId2,
+                "0",
+                "1000000000000"
+            );
+
+            const updateRewardSpeed = api.tx.sudo.sudo(
+                sudoInterBtcAPI.api.tx.utility.batchAll([updateRewardSpeedExtrinsic_1, updateRewardSpeedExtrinsic_2])
+            );
+
+            const rewardExtrinsic = sudoInterBtcAPI.api.tx.utility.batchAll([addRewardExtrinsic, updateRewardSpeed]);
+
+            const result = await DefaultTransactionAPI.sendLogged(
+                api,
+                sudoAccount,
+                rewardExtrinsic,
+                api.events.sudo.Sudid
+            );
+
+            expect(result.isCompleted, "Sudo event to add rewards not found").to.be.true;
+        });
+
+        it("should return correct amount of rewards", async () => {
+            await submitExtrinsic(
+                userInterBtcAPI,
+                await userInterBtcAPI.loans.lend(underlyingCurrency, newMonetaryAmount(1, underlyingCurrency, true)),
+                false
+            );
+
+            const rewards = await userInterBtcAPI.loans.getAccruedRewardsOfAccount(userAccountId);
+
+            expect(rewards.total.toBig().eq(1)).to.be.true;
```

Comment:
You can also lock the collateral and borrow `underlyingCurrencyId2` in the same test, to ensure the borrow reward is correctly calculated as well

Reply:
Good idea, thanks)